frontend/packages/core-api-client/src/client/Recipient.ts

```diff
@@ -64,8 +64,23 @@ export class RecipientClient implements RecipientClientDefinition {
     );
   };
 
-  public list = (): Promise<RecipientList> => {
-    return Promise.resolve({ items: [] });
+  public list = async ({
+    formId,
+    databaseId,
+  }: FormLookup): Promise<FormWithRecord<Recipient>[][]> => {
```

Comment:
I think this change is worth doing outside of this pr as a refactor later.

I'm thinking `FormWithRecord<Recipient>[][]` isn't the best return type here as we're duplicating a lot of data as the forms are the same for each record. We could probably reduce the api calls made by doing a `list` for each form and then connecting the records together.

Something like:
```
{
  forms: FormDefinition[];
  records: Record<string, Record>[]; // Different Record types
}
```
or
```
{
  forms: FormDefinition[];
  records: Record[][];
}
```

or maybe `FormWithRecord<Recipient>[][]` is the best return type as it's used by the frontend widely, but how we build it could be changed.

Reply:
yeah, the current structure is definitely not ideal, we should do this when we find time